tests/prescribed_temperature_in_field.cc

```diff
@@ -0,0 +1,141 @@
+/*
+  Copyright (C) 2011 - 2020 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+#include <deal.II/base/parameter_handler.h>
+#include <deal.II/base/parsed_function.h>
+#include <deal.II/fe/fe_values.h>
+#include <aspect/global.h>
+#include <aspect/utilities.h>
+#include <aspect/simulator_signals.h>
+#include <aspect/initial_temperature/interface.h>
+
+namespace aspect
+{
+  using namespace dealii;
+
+  // Declare and parse additional parameters
+  bool prescribe_internal_temperature;
+  unsigned int fixed_composition_idx;
+  void declare_parameters(const unsigned int dim,
+                          ParameterHandler &prm)
+  {
+    prm.declare_entry ("Prescribe internal temperature", "false",
+                       Patterns::Bool (),
+                       "Whether or not to use any prescribed internal temperature. "
+                       "Locations in which to prescribe temperature are defined "
+                       "in section ``Prescribed temperature/Indicator function'' "
+                       "and the temperature are defined in section ``Prescribed "
+                       "temperature/Temperature function''. Indicators are evaluated "
+                       "at the center of each cell, and all DOFs associated with "
+                       "the specified temperature at the indicated cells "
+                       "are constrained."
+                      );
+    prm.declare_entry ("Index of fixed compositional field", "0",
```

Comment:
Can you make this a list of integers (`Patterns::List(Patterns::Integer`) so that one can fix the temperature in more than one compositional field?